src/daft-sql/src/planner.rs

```diff
@@ -900,56 +926,69 @@ impl SQLPlanner {
         }
     }
 
-    fn plan_compound_identifier(&self, idents: &[Ident]) -> SQLPlannerResult<Vec<ExprRef>> {
-        let ident_str = idents_to_str(idents);
-        let mut idents = idents.iter();
-
-        let root = idents.next().unwrap();
-        let root = ident_to_str(root);
-        let current_relation = match self.get_table_from_current_scope(&root) {
-            Some(rel) => rel,
-            None => {
-                return Err(PlannerError::TableNotFound {
-                    message: "Expected table".to_string(),
-                })
-            }
-        };
+    /// Find the ident information in the current query or parent query if in a subquery
+    fn find_ident_in_queries(&self, idents: &[Ident]) -> SQLPlannerResult<OuterReferenceColumn> {
+        let full_str = idents_to_str(idents);
+        let current_relation = self.relation_opt().unwrap();
+
+        // If the entire identifier is present in the schema, we are done.
+        // This also covers duplicate columns from joins, which are added to the table with a prefix (df.column_name)
+        if let Ok(field) = current_relation.schema().get_field(&full_str) {
+            return Ok(OuterReferenceColumn {
+                field: field.clone(),
+                depth: 0,
+            });
+        }
 
-        if root == current_relation.get_name() {
-            // This happens when it's called from a qualified wildcard (tbl.*)
-            if idents.len() == 0 {
-                return Ok(current_relation
-                    .inner
-                    .schema()
-                    .fields
-                    .keys()
-                    .map(|f| col(f.clone()))
-                    .collect());
-            }
```

Comment:
Didn't port over this logic because I don't believe that a qualified wildcard will hit this path (it will instead match on `SQLExpr::QualifiedWildcard`)